platform/pulp/platform/models/task.py

```diff
@@ -0,0 +1,296 @@
+
+    start_time = models.DateTimeField()
+    finish_time = models.DateTimeField()
+
+    _error = models.TextField(null=True, db_column='error')
+    _result = models.TextField(null=True, db_column='result')
+
+    parent = models.ForeignKey("Task", null=True)
+    worker = models.ForeignKey(Worker)
+
+    @property
+    def result(self):
+        try:
+            return self._result_object
+        except AttributeError:
+            self._result_object = json.loads(self._result)
+            return self._result_object
+
+    @property
+    def error(self):
+        try:
+            return self._error_object
+        except AttributeError:
+            self._error_object = json.loads(self._result)
+            return self._error_object
+
+
+    def save(self, *args, **kwargs):
+        try:
+            self._args = json.dumps(self._error_object)
+        except AttributeError:
+            pass
+
+        try:
+            self._kwargs = json.dumps(self._result_object)
+        except AttributeError:
+            pass
+
+        super(Model, self).save(*args, **kwargs)
+
+
+class TaskTags(Model):
+    """
+    Custom tags for a task
+
+    Fields:
+
+    :cvar name: The name of the tag
+    :type name: models.TextField
+
+    Relations:
+
+    :cvar task: The task this tag is associated with
+    :type task: models.ForeignKey
+    """
+    name = models.TextField()
+
+    task = models.ForeignKey(Task, related_name="tags", related_query_name="tag")
+
+class ScheduledCalls(Model):
+    """
+    Scheduled Call Request
+
+    Fields:
+
+    :cvar task: The task that should be run on a schedule
+    :type task: models.TextField
+
+    :cvar enabled: Indicates if schedule should be actively run by the scheduler
+    :type enabled: models.BooleanField
+
+    :cvar resource: Indicates a unique resource that should be used to find this schedule
+    :type resource: models.TextField
+
+    :cvar consecutive_failures: Number of times this task has failed consecutively
+    :type consecutive_failures: models.IntegerField
+
+    :cvar failure_threshold: Number of consecutive failures before task is automatically disabled
+    :type failure_threshold: models.IntegerField
+
+    :cvar iso_schedule: ISO8601 string representing the schedule
+    :type iso_schedule: models.TextField
+
+    :cvar schedule: Pickled instance of celery.schedules.schedule that should be run.
+    :type schedule: models.TextField
+
+    :cvar first_run: The first time this schedule was ran
+    :type first_run: models.DateTimeField
+
+    :cvar remaining_runs: Number of runs left until this schedule gets disabled
+    :type remaining_runs: models.IntegerField
+
+    :cvar total_run_count: Number of times this schedle has ran
+    :type total_run_count: models.IntegerField
+
+    :cvar last_run_at: Last time this schedule was ran
+    :type last_run_at: models.DateTimeField
+
+    :cvar last_updated: The last time this schedule was saved to the database
```

Comment:
I think this is specifically the last time a user changed something about the schedule. This field helps the ``Scheduler`` know if it needs to re-load schedules from the database when considering whether any are due to be run. I do not think this field should be touched when something like the ``total_run_count`` is incremented.